test/spmd/tensor/test_dtensor_ops.py

```diff
@@ -0,0 +1,875 @@
+    aten.zeros_like.default: {f32},
+}
+
+# these sometimes pass and sometimes fail
+# i.e. view only works with certain sharding dims
+dtensor_dispatch_skips = {
+    aten.view.default: {f32},
+    aten.unsqueeze.default: {f32},
+    aten.repeat.default: {f32},
+    aten.cat.default: {f32},
+}
+
+dtensor_dispatch_device_expected_failures = defaultdict(dict)
+dtensor_dispatch_device_skips = defaultdict(dict)
+
+# ops inside this might even fail without dtensor
+# tests, as we rescale op db common test size factor (i.e. L, M, S)
+# which triggered the orignal function run failures with input
+# generation becomes wrong, we skip them for now but should enable later.
+# TODO: need to clean this list and remove all cases
+op_inputs_skips = {
+    "argwhere",
+    "__rmatmul__",
+    "softmax",
+    "nn.functional.softmin",
+    "nn.functional.embedding",
+    "nn.functional.feature_alpha_dropout",
+    "nn.functional.hinge_embedding_loss",
+}
+
+
+class DTensorCrossRefDispatchMode(
+    torch.utils._python_dispatch.TorchDispatchMode
+):
+    test_case: TestCase
+    device: torch.device
+    dtype: torch.dtype
+
+    def __init__(self, test_case, *, device, dtype):
+        self.test_case = test_case
+        # save TLS
+        self.precision = test_case.precision
+        self.rel_tol = test_case.rel_tol
+        self.device_type = torch.device(device).type
+        self.dtype = dtype
+
+    def __torch_dispatch__(self, func, types, args=(), kwargs=None):
+        dispatch_functions.setdefault(func, set()).add(self.dtype)
+        kwargs = kwargs or {}
+
+        self.test_case.precision = self.precision
+        self.test_case.rel_tol = self.rel_tol
+
+        if self.dtype in dtensor_dispatch_skips.get(func, set()):
+            test_expect = TestExpect.SKIP
+        elif self.dtype in dtensor_dispatch_device_skips[self.device_type].get(
+            func, set()
+        ):
+            test_expect = TestExpect.SKIP
+        elif self.dtype in dtensor_dispatch_expected_failures.get(func, set()):
+            test_expect = TestExpect.XFAILURE
+        elif self.dtype in dtensor_dispatch_device_expected_failures[
+            self.device_type
+        ].get(func, set()):
+            test_expect = TestExpect.XFAILURE
+        else:
+            test_expect = TestExpect.SUCCESS
+
+        return run_dtensor_crossref(
+            self.test_case,
+            test_expect,
+            func,
+            args,
+            kwargs,
+            dtype=self.dtype,
+            device_type=self.device_type,
+        )
+
+
+class TestDTensorOps(DistTensorTestBase):
+    @property
+    def world_size(self) -> int:
+        return 4
+
+    # only allow float dytpe for now, we can relax this constraint
+    # when feel necessary later (i.e when adding quantization support).
+    @unittest.skipIf(TEST_WITH_ASAN, "Skipped under ASAN")
+    @skipIfCrossRef
+    @suppress_warnings
+    @ops(common_ops.op_db, allowed_dtypes=(torch.float,))
```

Comment:
Do they regularly sync this list or it's automatically generated?

Reply:
it's op db from pytorch itself, so operators stored in pytorch tests will be synced regularly if pytorch op tests get updated.